pysidt/sidt.py

```diff
@@ -932,6 +946,529 @@ def regularize(self, data=None, check_data=True):
+            for i, d in enumerate(self.mol_node_maps[datum]["mols"]):
+                if any(d is x for x in new):
+                    assert d.is_subgraph_isomorphic(
+                        node.group, generate_initial_map=True, save_order=True
+                    )
+                    self.mol_node_maps[datum]["nodes"][i] = node
+
+        logging.info("adding node {}".format(name))
+        
+        if grpc:
+            class_true = 0
+            frags = name.split("_")
+            frags[-1] = "N-" + frags[-1]
+            cextname = ""
+            for k in frags:
+                cextname += k
+                cextname += "_"
+            cextname = cextname[:-1]
+            nodec = Node(
+                group=grpc,
+                items=comp,
+                rule=comp_rule,
+                parent=parent,
+                children=[],
+                name=cextname,
+                depth=parent.depth + 1,
+            )
+
+            self.nodes[cextname] = nodec
+            parent.children.append(nodec)
+            self.new_nodes.append(cextname)
+
+            for k, datum in enumerate(self.datums):
+                for i, d in enumerate(self.mol_node_maps[datum]["mols"]):
+                    if any(d is x for x in comp):
+                        assert d.is_subgraph_isomorphic(nodec.group, generate_initial_map=True, save_order=True), (d.to_adjacency_list(),nodec.group.to_adjacency_list())
+                        self.mol_node_maps[datum]["nodes"][i] = nodec
+
+            parent.items = []
+        else:
+            parent.items = comp
+            parent.rule = comp_rule
+
+    def evaluate(self, mol):
+        """
+        Evaluate tree for a given possibly labeled mol
+        """
+        out = 0.0
+        decomp = self.decomposition(mol)
+        for d in decomp:
+            children = self.root.children
+            node = self.root
+            boo = True
+            while boo:
+                for child in children:
+                    if d.is_subgraph_isomorphic(
+                        child.group, generate_initial_map=True, save_order=True
+                    ):
+                        children = child.children
+                        node = child
+                        break
+                else:
+                    boo = False
+
+            if not node.rule:
+                return False
+                    
+        return True
+
+    def analyze_error(self):
+        """
+        compute overall training and validation errors
+        """
+        sidt_train_values = [self.evaluate(d.mol) for d in self.datums]
+        true_train_values = [d.value for d in self.datums]
+        if self.validation_set:
+            sidt_val_values = [self.evaluate(d.mol) for d in self.validation_set]
+            true_val_values = [d.value for d in self.validation_set]
+
+        P,N,PP,PN,TP,FN,FP,TN = analyze_binary_classification(sidt_train_values,true_train_values)
```

Comment:
I'm pretty sure there are metrics functions to compute accuracy, recall, precision, etc. we can just import from sklearn to make the code cleaner. Can you do that?

```suggestion
        P, N, PP, PN, TP, FN, FP, TN = analyze_binary_classification(sidt_train_values, true_train_values)
```